python/src/equistore/operations/unique_metadata.py

```diff
@@ -0,0 +1,253 @@
+    Finds the unique metadata of a list of blocks along the given ``axis`` and
+    for the specified ``names``. If ``gradient_param`` is specified, only finds
+    the unique indices for gradient blocks under the specified parameter name.
+    """
+    # Extract indices from each block
+    all_idxs = []
+    for block in blocks:
+        if axis == "samples":
+            idxs = block.samples[names]
+        else:  # "properties"
+            idxs = block.properties[names]
+        all_idxs += idxs.tolist()
+
+    # If no matching indices across all blocks return a empty Labels w/ the
+    # correct names
+    if len(all_idxs) == 0:
+        # Create Labels with single entry
+        labels = Labels(names=names, values=np.arange(len(names)).reshape(-1, 1))
+        return labels[:0]  # Slice to zero length
+
+    # Define the unique and sorted indices
+    unique_idxs = np.unique(all_idxs, axis=0)
+
+    # Return as Labels
+    return Labels(names=names, values=np.array([[j for j in i] for i in unique_idxs]))
+
+
+def _check_args(
+    tensor: Union[TensorMap, TensorBlock],
+    axis: str,
+    names: List[str],
+    gradient_param: Optional[str] = None,
+):
+    """
+    Checks input args for :py:func:`unique_metadata` and
+    :py:func:`unique_metadata_block`.
+    """
+    # Check tensors
+    if isinstance(tensor, TensorMap):
+        blocks = tensor.blocks()
+        # Check gradients
+        if gradient_param is not None:
+            if not isinstance(gradient_param, str):
+                raise TypeError("`gradient_param` must be a `str`")
+            # Check all blocks have a gradient under the passed param
+            if not np.all([block.has_gradient(gradient_param) for block in blocks]):
+                raise ValueError(
+                    "not all input blocks have a gradient under the"
+                    + f" passed `gradient_param` {gradient_param}"
+                )
+            blocks = [
+                block.gradient(gradient_param) for block in blocks
+            ]  # redefine blocks
+    elif isinstance(tensor, TensorBlock):
+        blocks = [tensor]
+        # Check gradients
+        if gradient_param is not None:
+            if not isinstance(gradient_param, str):
+                raise TypeError("`gradient_param` must be a `str`")
+            # Check block has a gradient under the passed param
+            if not tensor.has_gradient(gradient_param):
+                raise ValueError(
+                    "input block does not have a gradient under the"
+                    + f" passed `gradient_param` {gradient_param}"
+                )
+            blocks = [tensor.gradient(gradient_param)]  # redefine blocks
+    # Check axis
+    if not isinstance(axis, str):
+        raise TypeError("`axis` must be a `str`, either `'samples'` or `'properties'`")
+    if axis not in ["samples", "properties"]:
+        raise ValueError(
+            "`axis` must be passed as either `'samples'` or `'properties'`"
```

Comment:
```suggestion
            f"`axis` must be passed as either `'samples'` or `'properties'`, {axis} was passed."
```
or something like that here and in the other checks